src/components/send-money/SendMoneyAmountInput.js

```diff
@@ -55,58 +60,70 @@ const CustomDiv = styled(`div`)`
    }
 `
 
+const Big = require('big.js')
 class SendMoneyAmountInput extends Component {
    state = {
-      amount: `${this.props.defaultAmount}` || '',
+      amountInput: `${this.props.defaultAmount}` || '',
       amountStatus: ''
    }
 
+   isDecimalString = (value) => {
+      let REG = /^[0-9]*(|[.][0-9]{1,5})$/
+      return REG.test(value)
+   }
+
    handleChangeAmount = (e, { name, value }) => {
-      const amountStatus = !Number.isInteger(Number(value))
-         ? 'Please enter a whole number.'
-         : value > Number(this.props.amount)
-            ? 'Not enough tokens.' 
-            : ''
+      let amountStatus = ''
+      if (value && !this.isDecimalString(value)) {
+         amountStatus = 'NO MORE THAN 5 DECIMAL DIGITS'
+      }
 
+      if (value !== '') {
+         let input = new Big(value)
+         let balance = new Big(this.props.amount)
+         if (balance.sub(input).s < 0) {
+            amountStatus = 'Not enough tokens.'
+         }
+      }
       this.setState(() => ({
-         [name]: value,
+         amountInput: value,
          amountStatus
       }))
 
-      this.props.handleChange(e, { name, value })
+      this.props.handleChange(e, { name: 'amount', value })
       this.props.handleChange(e, { name: 'amountStatus', value: amountStatus })
    }
 
-   render () {
-      const { amount, amountStatus } = this.state
-      const fontSize = amount.length > 11 ? 32 : amount.length > 8 ? 38 : amount.length > 5 ? 50 : 72
+   render() {
+      const { amountInput, amountStatus } = this.state
+      const fontSize = amountInput.length > 11 ? 32 : amountInput.length > 8 ? 38 : amountInput.length > 5 ? 50 : 72
 
       return (
          <CustomDiv fontSize={`${fontSize}px`}>
             <Form.Input
                type='number'
-               pattern='[0-9]*'
-               name='amount'
-               value={amount}
+               name='amountInput'
+               value={amountInput}
                onChange={this.handleChangeAmount}
                placeholder='0'
                step='1'
                min='1'
                tabIndex='2'
+               required={true}
             />
             {amountStatus && (
                <Segment basic textAlign='center' className='alert-info problem'>
                   {amountStatus}
-               </Segment>
-            )}
+               </Segment>)}
+            {amountInput ? <Balance milli={milli} amount={amountInput} /> : "How much would you want to send? Please send at least 10,000,000,000 "}
          </CustomDiv>
       )
    }
 }
 
 SendMoneyAmountInput.propTypes = {
    handleChange: PropTypes.func.isRequired,
-   amount: PropTypes.string
+   amountInput: PropTypes.string
```

Comment:
why rename?

Reply:
`amount` here means walletAccount.amount, and there is also local state used to name amount, I change that to amountInput